spec/reek/source/source_locator_spec.rb

```diff
@@ -0,0 +1,42 @@
+require_relative '../../spec_helper'
+require_relative '../../../lib/reek/source/source_locator'
+
+RSpec.describe Reek::Source::SourceLocator do
+  describe '#sources' do
+    let(:path) { 'spec/samples/two_smelly_files' }
+
+    it 'returns a list of Source::SourceCode with the right paths' do
+      sources = described_class.new([path]).sources
+
+      expect(sources.all? { |source| source.is_a? Reek::Source::SourceCode }).to be true
+      expect(sources.map(&:description).sort).to eq [
+        'spec/samples/two_smelly_files/dirty_one.rb',
+        'spec/samples/two_smelly_files/dirty_two.rb'
+      ]
+    end
+
+    context 'exclude paths' do
+      let(:config) { 'spec/samples/configuration/with_excluded_paths.reek' }
+      let(:path) { 'spec/samples/source_with_exclude_paths' }
+      let(:files_that_are_expected_to_be_ignored) do
+        [
+          'spec/samples/source_with_exclude_paths/ignore_me/uncommunicative_method_name.rb',
+          'spec/samples/source_with_exclude_paths/nested/ignore_me_as_well/irresponsible_module.rb'
+        ]
+      end
+
+      it 'does not use excluded paths' do
+        with_test_config(config) do
+          sources = described_class.new([path]).sources
+
+          expect(sources.map(&:description).sort).
```

Comment:
I know that this is kind of implicitly checked by the expectation below but this makes it more palpable for the innocent reader.